app/javascript/mastodon/components/status_action_bar.js

```diff
@@ -57,6 +58,13 @@ export default class StatusActionBar extends ImmutablePureComponent {
     this.props.onReply(this.props.status, this.context.router.history);
   }
 
+  handleShareClick = () => {
+    navigator.share({
+      text: this.props.status.get('search_index'),
```

Comment:
Not sure about the usage of search_index for this purpose. It's both content warning and proper text, yes.. But it's purpose is to be used in client-side search. We might make it do fancier indexing stuff later and then this usage will be out of date.